src/test/scala/com/databricks/labs/overwatch/pipeline/InitializeTest.scala

```diff
@@ -0,0 +1,224 @@
+
+      assert(intelligentScaling == actualIntelligentScaling)
+    }
+  }
+
+  describe("Tests for quickBuildAuditLogConfig configs") {
+
+    it("quickBuildAuditLogConfig function should remove the last / from audit log path") {
+      import com.databricks.labs.overwatch.utils.AuditLogConfig
+
+      val configInput = AuditLogConfig(Some("path/to/auditLog/"), "Json", None)
+      val conf = new Config
+      val init = new Initializer(conf)
+      val quickBuildAuditLogConfig = PrivateMethod[AuditLogConfig]('quickBuildAuditLogConfig)
+
+      val expectedAuditConf = AuditLogConfig(Some("path/to/auditLog"), "json", None)
+      val actualAuditConf = init invokePrivate quickBuildAuditLogConfig(configInput)
+      assert(expectedAuditConf == actualAuditConf)
+    }
+
+    it("quickBuildAuditLogConfig function build path from prefix for azure eventhub config") {
+      import com.databricks.labs.overwatch.utils.{AuditLogConfig, AzureAuditLogEventhubConfig}
+
+      val configInput = AuditLogConfig(None, "json", Some(AzureAuditLogEventhubConfig("sample.connection.string","auditLog",
+        "path/to/auditLog/prefix/",10000,10, None, None)))
+
+      val conf = new Config
+      val init = new Initializer(conf)
+      val quickBuildAuditLogConfig = PrivateMethod[AuditLogConfig]('quickBuildAuditLogConfig)
+
+      val expectedAuditConf = AuditLogConfig(None, "json", Some(AzureAuditLogEventhubConfig("sample.connection.string",
+        "auditLog", "path/to/auditLog/prefix",10000,10, Some("path/to/auditLog/prefix/rawEventsCheckpoint"), Some("path/to/auditLog/prefix/auditLogBronzeCheckpoint"))))
+      val actualAuditConf = init invokePrivate quickBuildAuditLogConfig(configInput)
+      assert(expectedAuditConf == actualAuditConf)
+    }
+
+
+  }
+
+  describe("Tests for validateAuditLogConfigs configs") {
+
+    it("validateAuditLogConfigs function validate auditLogPath in the config ") {
+      import com.databricks.labs.overwatch.utils.AuditLogConfig
+
+      val configInput = AuditLogConfig(None, "Json", None)
+      val conf = new Config
+      conf.setCloudProvider("aws")
+      val init = new Initializer(conf)
+      val quickBuildAuditLogConfig = PrivateMethod[AuditLogConfig]('validateAuditLogConfigs)
+
+      assertThrows[BadConfigException](init invokePrivate quickBuildAuditLogConfig(configInput))
+    }
+
+    it("validateAuditLogConfigs function validate audit log format in the config ") {
+      import com.databricks.labs.overwatch.utils.AuditLogConfig
+
+      val configInput = AuditLogConfig(Some("path/to/audit/log"), "text", None)
+      val conf = new Config
+      conf.setCloudProvider("aws")
+      val init = new Initializer(conf)
+      val quickBuildAuditLogConfig = PrivateMethod[AuditLogConfig]('validateAuditLogConfigs)
+
+      assertThrows[BadConfigException](init invokePrivate quickBuildAuditLogConfig(configInput))
+    }
+  }
+
+
+  describe("Tests for validateAndRegisterArgs function") {
```

Comment:
Some additional tests are needed.

- test for missing parameters
- test for duplicate parameters
- test for non-escaped json
- test for malformed json